### ISSUE TITLE
curl CLI cannot take credentials from Windows Credential Manager

### ISSUE THREAD

**Issue by escherstair** [curl/curl#13011]
### I did this

I found [this discussion](https://superuser.com/questions/250053/how-can-i-pass-my-windows-credentials-in-curl) because I need a way to have curl CLI taking the proper credentials from the Windows Credential Manager on Windows 10 machine.
I downloaded the binary for Windows [https://curl.se/windows/dl-8.6.0_2/curl-8.6.0_2-win64-mingw.zip](https://curl.se/windows/dl-8.6.0_2/curl-8.6.0_2-win64-mingw.zip).
I run
`curl --ntlm -u : ...`

### I expected the following

I would expect seeing this message
`Server auth using NTLM with user 'username_from_credential_manager'`
but I see
`Server auth using NTLM with user ''`
So, empty user and 
`HTTP/1.1 401 Unauthorized`

If I run
`curl --ntlm -u : username_from_credential_manager:password_from_credential_manager`
but in this way the credentials are passed to the command line not encrypted
I see
```
* Server auth using NTLM with user 'username_from_credential_manager'
> POST /ws/foo/v2/barHTTP/1.1
> Host: aaaaaaa.bbbb.hhhh.com:55451
> Authorization: NTLM TlRMTVNTUAABAAAABoIIAAAAAAAAAAAAAAAAAAAAAAA=
> User-Agent: curl/8.6.0
> Content-Type: application/json
> Accept: application/json
> Content-Length: 0
```

But, see that 
`> Content-Length: 0`
and this is not true, sice a pass a json file through `-d @file.json`

But then I get
```
< HTTP/1.1 401 Unauthorized
< WWW-Authenticate: Basic Realm = 'Integration Server'
< Content-Type: application/json; charset=UTF-8
< Content-Length: 308
```

If I remove `--ntlm` I get a basic authorization attempt and I read
`> Content-Length: 7`
that is fine
 
### curl/libcurl version

curl 8.6.0 (x86_64-w64-mingw32) libcurl/8.6.0 LibreSSL/3.8.2 zlib/1.3.1 brotli/1.1.0 zstd/1.5.5 WinIDN libpsl/0.21.5 libssh2/1.11.0 nghttp2/1.59.0 ngtcp2/1.3.0 nghttp3/1.2.0
Release-Date: 2024-01-31
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns ldap ldaps mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp ws wss
Features: alt-svc AsynchDNS brotli HSTS HTTP2 HTTP3 HTTPS-proxy IDN IPv6 Largefile libz NTLM PSL SSL threadsafe UnixSockets zstd

### operating system

Windows 10

**Comment by bagder** [comment]
Thanks, but this description sounds as if you're asking for a new feature/change. We use this tracker for bugs and issues only, we put ideas to work on in the future in the TODO document. We basically drown in good ideas so they don't do much use in our tracker.

If you really want to see this happen, start working on an implementation and submit a PR for it or join the mailing list and talk up more interest for it and see what help from others you can get!

**Comment by escherstair** [comment]
Hi @bagder 
thank you very much for your answer.
English is not my native language, so I'm not sure I understood.

Do you mean that at the moment curl cannot take credentials from Windows Credential Manager (based on the server url)?
And so I must supply username and password explicitly?
If this is the case, I thought it was an existing feature and for this reason I opened the issue.
Sorry for that.

Thanks

**Comment by bagder** [comment]
I don't even know what Windows Credential Manager is. But now I also see that the curl you use is not built with SSPI support, which I now think might be what otherwise would do this magic. Perhaps.

**Comment by escherstair** [comment]
Thanks a lot @bagder;
your explanation has been really useful for me.
I've been able to solve.
I write here what I learned, in case it's useful for someone else:

`curl --ntlm -u : ...` (that passes an empty `username:password`) does the trick (i.e., takes the right credentials from Windows Credential Manager automatically), but only if you use [curl shipped by Microsoft](https://curl.se/windows/microsoft.html).
This is built with SSPI support, and some other compiler options.

The [Windows build](https://curl.se/windows/) offered by curl project, lacks of SSPI support (and maybe some other MS-specific features) and I haven't been able to see it working in any way.

Long-story-short: use [curl shipped by Microsoft](https://curl.se/windows/microsoft.html) and run `curl --ntlm -u : ...` so that the proper credentials from WCM (Windows Credentials Manager) are used automatically.

I think this issue can be closed.
Thanks a lot.
